pkg/log/log.go

```diff
@@ -150,6 +151,29 @@ func format(msg, level string, ts time.Time, appInfo Marshaler, mm Many) string
 		generateFatalFields(entry)
 	}
 
+	// Attempt to map the caller of the log function into the "source" field for identifying if a service or a module
+	// that the service is using is sending logs (costing money).
+	caller, err := callerinfo.GetCallerFunction(2)
+	switch {
+	case err != nil:
+		entry["source"] = "error"
+	case strings.HasPrefix(caller, "github.com/"):
+		// Example response: github.com/getoutreach/gobox/pkg/callerinfo.Test_Callers
+		splits := strings.Split(caller, "/")
+		// We're going for extracting "getoutreach/gobox" in the above example
+		if len(splits) >= 3 {
+			entry["source"] = splits[1] + "/" + splits[2]
```

Comment:
I would not strip off `github.com`, as it's part of the module name.

Reply:
I really wish we had a way to get the canonical module name from https://pkg.go.dev/runtime/debug#ReadBuildInfo...

Reply:
Noted this in a DM as well, but we could use the aforementioned build information to retrieve the version of the library in use and store it in a separate field like `source_version`.

Reply:
done -- reworked the whole thing